src/ServicePulse.Host/vue/src/components/failedmessages/AllFailedMessages.vue

```diff
@@ -11,63 +11,58 @@ import { useArchiveExceptionGroup, useRetryExceptionGroup } from "../../composab
 import LicenseExpired from "../../components/LicenseExpired.vue";
 import OrderBy from "./OrderBy.vue";
 import ServiceControlNotAvailable from "../ServiceControlNotAvailable.vue";
-import MessageList from "./MessageList.vue";
+import MessageList, { IMessageList } from "./MessageList.vue";
 import ConfirmDialog from "../ConfirmDialog.vue";
 import PaginationStrip from "../../components/PaginationStrip.vue";
-import { FailedMessageStatus } from "@/resources/FailedMessage";
+import { ExtendedFailedMessage, FailedMessageStatus } from "@/resources/FailedMessage";
+import SortOptions, { SortDirection } from "@/resources/SortOptions";
+import { TYPE } from "vue-toastification";
 
 let pollingFaster = false;
-let refreshInterval = undefined;
-let sortMethod = undefined;
+let refreshInterval: number | undefined;
+let sortMethod: SortOptions | undefined;
 const perPage = 50;
 const route = useRoute();
-const groupId = ref(route.params.groupId);
+const groupId = ref<string>(route.params.groupId as string);
 const groupName = ref("");
 const pageNumber = ref(1);
 const totalCount = ref(0);
 const showDelete = ref(false);
 const showConfirmRetryAll = ref(false);
 const showConfirmDeleteAll = ref(false);
-const messageList = ref();
-const messages = ref([]);
-const sortOptions = [
+const messageList = ref<IMessageList>();
+const messages = ref<ExtendedFailedMessage[]>([]);
+const sortOptions: SortOptions[] = [
   {
     description: "Time of failure",
-    selector: function (group) {
```

Comment:
note that this logic was *wrong*, but didn't matter because selector is only used for client-side sorting